src/PSRule/Pipeline/Output/JsonOutputWriter.cs

```diff
@@ -22,16 +27,50 @@ protected override string Serialize(object[] o)
                 NullValueHandling = NullValueHandling.Ignore
             };
 
-            int? outputJsonIndent = Option.Output.JsonIndent;
-            if (outputJsonIndent.HasValue && outputJsonIndent > 0)
+            var outputJsonIndent = Option.Output.JsonIndent ?? 0;
+            var isIndented = outputJsonIndent > 0;
+
+            if (isIndented)
             {
                 jsonSerializer.Formatting = Formatting.Indented;
-                jsonTextWriter.Indentation = outputJsonIndent.Value;
+                jsonTextWriter.Indentation = outputJsonIndent;
             }
 
-            jsonSerializer.ContractResolver = new SortedPropertyContractResolver();
+            jsonSerializer.ContractResolver = new OrderedPropertiesContractResolver();
 
             jsonSerializer.Converters.Add(new ErrorCategoryJsonConverter());
+            jsonSerializer.Converters.Add(new PSObjectJsonConverter());
+
+            // To avoid writing baselines with an extra outer array
+            // We can serialize the first object which has all the baselines
+            if (o[0] is IEnumerable<Baseline> baselines)
+            {
+                jsonSerializer.Converters.Add(new BaselineConverter());
+
+                jsonSerializer.Serialize(jsonTextWriter, baselines);
+
+                var indentMultiplier = string.Concat(Enumerable.Repeat(" ", outputJsonIndent * 2));
+
+                var jsonString = stringWriter.ToString();
+
+                // Workaround given JSON.NET doesn't support writing inline comments
+                // Can write multiline comments first then replace with inline comment delimiter
+                // We use /\*(.*?)\*/ to capture comments between /* */ deimiters
+                var jsonWithMultilineCommentsReplaced = Regex.Replace(jsonString, @"/\*(.*?)\*/", match =>
```

Comment:
This was the workaround I did to write `//` comments. The `JsonWriter.WriteComment` method in JSON.Net only writes multiline `/* */` comments. Let me know if you find a better way. This seems like a hack, but works I guess 😄.